packages/sdk/src/oracle/services/erc20.ts

```diff
@@ -0,0 +1,43 @@
+import { erc20 } from "../../clients";
+import Multicall2 from "../../multicall2";
+import { BatchReadWithErrors, BatchReadWithErrorsType, Calls } from "../../utils";
+import { BigNumber, Provider } from "../types/ethers";
+
+const batchProps: Calls = [["symbol"], ["name"], ["decimals"], ["totalSupply"]];
+type Props = {
+  symbol: string;
+  name: string;
+  decimals: number;
+  totalSupply: BigNumber;
+};
+export class Erc20 {
+  public contract: erc20.Instance;
+  constructor(protected provider: Provider, protected address: string) {
+    this.contract = erc20.connect(address, provider);
+  }
+  async getProps(): Promise<Props> {
+    const { contract } = this;
+    return {
+      symbol: await contract.callStatic.symbol(),
+      name: await contract.callStatic.name(),
+      decimals: await contract.callStatic.decimals(),
+      totalSupply: await contract.callStatic.totalSupply(),
```

Comment:
this would have been better as promise.all, but could not get typescript to pass. its slightly more complicated because of the override in the multicall class. so ended up with this solution

Reply:
What do you mean "could not get typescript to pass"? Any reason not to multicall these?

Reply:
I would have to go back and recreate the conditions to know the error, but typescript would keep complaining when i tried to convert an array of values back into an object of a certain type.